bundle/edu.gemini.itc.web/src/main/java/edu/gemini/itc/web/html/NifsPrinter.java

```diff
@@ -108,4 +125,23 @@ else if (instrument.getIFUMethod() instanceof IfuSummed)
         return s;
     }
 
+    public void getOverheadTableParams(Result result, ObservationDetails obs) {
+        int numberExposures = 1;
+        final CalculationMethod calcMethod = obs.calculationMethod();
+
+        if (calcMethod instanceof ImagingInt) {
+            numberExposures = (int)(((ImagingResult) result).is2nCalc().numberSourceExposures() * obs.sourceFraction());
+        } else if (calcMethod instanceof ImagingS2N) {
+            numberExposures = ((ImagingS2N) calcMethod).exposures();
+        } else if (calcMethod instanceof SpectroscopyS2N) {
+            numberExposures = ((SpectroscopyS2N) calcMethod).exposures();
+        }
+
+        ConfigCreator cc    = new ConfigCreator(p);
+        config              = cc.createNifsConfig(instr, numberExposures);
+        pta                 = PlannedTimeCalculator.instance.calc(config, new InstNIFS());
+
+        if (numberExposures < 2) step = 0; else step = 1;
+    }
+
```

Comment:
This `getOverheadTableParams` method seems repetitive from one `PrinterBase` to the next.

I would instead define it in `PrinterBase` as follows:
```java
public void getOverheadTableParams(Result result, ObservationDetails obs) {
         final int numberExposures;
         final CalculationMethod calcMethod = obs.calculationMethod();
 
         if (calcMethod instanceof ImagingInt) {
             numberExposures = (int)(((ImagingResult) result).is2nCalc().numberSourceExposures() * obs.sourceFraction());
         } else if (calcMethod instanceof ImagingS2N) {
             numberExposures = ((ImagingS2N) calcMethod).exposures();
         } else if (calcMethod instanceof SpectroscopyS2N) {
             numberExposures = ((SpectroscopyS2N) calcMethod).exposures();
         } else {
             numberExposures = 1;
         }
         config            = createInstConfig(numberExposures);
         pta                 = PlannedTimeCalculator.instance.calc(config, getInst());
         if (numberExposures < 2) step = 0; else step = 1;
     }

abstract Config[] createInstConfig(int numberExposures);

abstract SPInstObsComp getInst();
```

And then define just the different code of `createInstConfig` and `getInst()` in subclasses, e.g. here:

```java
Config[] createInstConfig(int numberExposures) {
  ConfigCreator cc = new ConfigCreator(p);
  return cc.createNifsConfig(instr, numberExposures);
}

SPInstObsComp getInst() {
  return new InstNIFS();
}
```

That way you don't have to write the same code over and over again, which is very unmaintainable.